### DISCUSSION TITLE
OIDC Discovery functionality

### DISCUSSION THREAD

**Discussion by simongottschlag** [go-kit/kit#1174] [General]
Hi!

I've been working on an extension for echo's JWT middleware to enable discovery through OpenID Connect. Everything using the jwx library.

Would you be interested in a PR for the same thing here?

You can see the PR here: https://github.com/labstack/echo-contrib/pull/59

If you are interested, I'd love some feedback on how you would prefer the API to be.

Thanks!

Regards,
Simon

**Comment by peterbourgon** [comment]
Is there anything about this package that would require it to live in Go kit?

**Reply by simongottschlag** [reply]
Hi @peterbourgon!

From my side, no - but I saw that you had the JWT auth middleware already and thought it would be a good fit to extend kit to also support OIDC out of the box.

But that would be up to you. Would you prefer it somewhere else?

**Reply by peterbourgon** [reply]
So in general Go kit has bitten off more than it can chew in terms of packages. Contributors contribute their PR and then disappear and then I have to maintain it 😉 and this is a mistake I'm trying to fix. Unless a package needs to exist in Go kit for technical or structural reasons, I'm basically saying "no thanks" from here on out. I'm also, slowly, removing existing packages, or at least moving them to other repos/modules. See #843 for an overview.

Please build this thing! Tell me when it's done and I'll link to it :)

**Reply by simongottschlag** [reply]
Fully understandable!

I'm going to rethink if there's a single place where I can place this code to make it available for as many frameworks as possible.

I would like to provide the functionality for at least kit, echo, mux (and standard http).

Maybe better if I create a go-oidc-middleware with the main logic and then packages wrapping it for each project.

Any ideas? Have you seen anything like that before?

**Reply by simongottschlag** [reply]
Hi @peterbourgon!

I've now created the following repository: https://github.com/XenitAB/go-oidc-middleware

My next step will be to make it compatible with kit and test/document it. I'll get back to you!

**Reply by simongottschlag** [reply]
After looking into kit deeper, I think it's not the same use case as the others. I can see that mux is being used for http and that means that it would be supported through it.

Sorry for taking up your time!
